executor/executor.go

```diff
@@ -1793,26 +1808,30 @@ func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
 	if priority := mysql.PriorityEnum(atomic.LoadInt32(&variable.ForcePriority)); priority != mysql.NoPriority {
 		sc.Priority = priority
 	}
-	if vars.StmtCtx.LastInsertID > 0 {
-		sc.PrevLastInsertID = vars.StmtCtx.LastInsertID
+	if saveLastInsertID > 0 {
+		sc.PrevLastInsertID = saveLastInsertID
 	} else {
-		sc.PrevLastInsertID = vars.StmtCtx.PrevLastInsertID
+		sc.PrevLastInsertID = savePrevLastInsertID
 	}
 	sc.PrevAffectedRows = 0
-	if vars.StmtCtx.InUpdateStmt || vars.StmtCtx.InDeleteStmt || vars.StmtCtx.InInsertStmt {
-		sc.PrevAffectedRows = int64(vars.StmtCtx.AffectedRows())
-	} else if vars.StmtCtx.InSelectStmt {
-		sc.PrevAffectedRows = -1
-	}
 	if globalConfig.EnableCollectExecutionInfo {
-		sc.RuntimeStatsColl = execdetails.NewRuntimeStatsColl()
+		// In ExplainFor case, RuntimeStatsColl should not be reset for reuse,
+		// because ExplainFor need to display the last statement information.
+		if _, ok := s.(*ast.ExplainForStmt); ok {
```

Comment:
Now I do not reset  `RuntimeStatsColl` when the statement is ExplainFor, the test passes.